### ISSUE TITLE
Uniformise seed and random state vocab

### ISSUE THREAD

**Issue by MarieSacksick** [probabl-ai/skore#2051]
### Is your feature request related to a problem? Please describe.

In some features, we use the word `seed`, such as in the [feature importance](https://github.com/probabl-ai/skore/blob/skore/0.11.0/skore/src/skore/_sklearn/_estimator/feature_importance_accessor.py#L310), and in others, such as in the [train test split](https://github.com/probabl-ai/skore/blob/skore/0.11.0/skore/src/skore/_sklearn/train_test_split/train_test_split.py#L25), we use word `random_state`.

### Describe the solution you'd like

Have something consistant.

### Additional context

I'd rather have random_state as it's what is used in scikit-learn.

**Comment by MarieSacksick** [comment]
I'd go for `random_state`, @glemaitre do you agree?

**Comment by divakaivan** [comment]
[There's this issue+PR which changed a lot of `random_state` into `seed`](https://github.com/probabl-ai/skore/issues/1397) by glemaitre earlier this year where he gave reasons for the changes

**Comment by MarieSacksick** [comment]
ooh, I see, thanks @divakaivan!  
I still think we should be consistant, therefore should we set the random_state of train test split to seed? Even though it can accept more than int theoretically, we could restrict to ints.

**Comment by glemaitre** [comment]
Thanks @divakaivan to find the discussion.

I would draw the line as follow:

- if it is a new `skore` object / function that is not a direct drop-in replacement with another library, then we should be consistent within `skore`.
- if it is a function or class where we aim to be a drop-in replacement from another library, like the `train_test_split` function for scikit-learn, I think it makes sense to preserve as much as possible the original signature. So it is up to the technical details such as random state and interaction with caching, for instance.
